routing.py

```diff
@@ -1,19 +1,39 @@
 import logging
 import urllib.parse as urlparse
+from enum import Enum
 from typing import Tuple
 
 from domain.metric import MetricType
 from domain.types import AgentID
 
-MAIN = "/"
-MATRIX = "/matrix"
-TIME_SERIES = "/time-series"
-
 logger = logging.getLogger("routing")
 
 
+class Route(Enum):
+    INDEX = "/"
+    MATRIX = "/matrix"
+    TIME_SERIES = "/time-series"
+    UNKNOWN = "[unknown_route]"
+
+
+def extract_route(pathname: str) -> Route:
+    """
+    Example:
+        pathname: /matrix?metric=Latency
+        return:   Route.MATRIX
+    """
+
+    route_end_index = pathname.find("?")
```

Comment:
_Non-critical, general comment:_

The CorrectSolution(tm) is to use `urllib.parse.urlparse`, like this:
```
from urllib.parse import urlparse
...
route_str = urlparse(pathname).path
```
The easy option is to use str.split (it does exactly what you need):
```
route_str = pathname.split("?")[0]
```